yarn-project/circuits.js/src/structs/read_request_membership_witness.ts

```diff
@@ -0,0 +1,125 @@
+import { Fr } from '@aztec/foundation/fields';
+import { assertMemberLength, range } from '../utils/jsUtils.js';
+import { serializeToBuffer } from '../utils/serialize.js';
+import { toBufferBE } from '@aztec/foundation/bigint-buffer';
+import { Tuple } from '@aztec/foundation/serialize';
+import { MAX_NEW_COMMITMENTS_PER_CALL, PRIVATE_DATA_TREE_HEIGHT } from './constants.js';
+import { MembershipWitness } from './membership_witness.js';
+
+/**
+ * A ReadRequestMembershipWitness is similar to a MembershipWitness but includes
+ * some additional fields used to direct the kernel regarding whether a read is transient
+ * and if so which commitment it corresponds to.
+ */
+export class ReadRequestMembershipWitness {
+  constructor(
+    /**
+     * Index of a leaf in the Merkle tree.
+     */
+    public leafIndex: bigint,
+    /**
+     * Sibling path of the leaf in the Merkle tree.
+     */
+    public siblingPath: Tuple<Fr, number>,
+    /**
+     * Does this read request correspond to a pending commitment?
```

Comment:
Re-use comment from cpp code. I do not like formulating a question for a description of a field. It sounds like the developer is not sure about sthg. (I understand the boolean is the response to the question though.)

Reply:
:+1: